pkg/syncer/syncer.go

```diff
@@ -23,41 +23,56 @@ import (
 	"strings"
 	"time"
 
+	coordinationv1 "k8s.io/api/coordination/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/clock"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/dynamic/dynamicinformer"
+	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
+	"k8s.io/component-helpers/apimachinery/lease"
 	"k8s.io/klog/v2"
 )
 
 const resyncPeriod = 10 * time.Hour
 const SyncerNamespaceKey = "SYNCER_NAMESPACE"
 
+// leaseRenewIntervalFraction is the fraction of lease duration to renew the lease
+const leaseRenewIntervalFraction = 0.25
+
+// syncerNamespace is the syncer namespace where cluster leases are created
+const syncerNamespace = "syncer-system"
+
 type Syncer struct {
+	stopCh       chan struct{}
```

Comment:
This has been refactored in main to use the ctx that's passed in to StartSyncer to control when to stop - please adjust accordingly. Thanks!